### PR TITLE
Document release process for editions

### LINKED ISSUE #279: Document our WoW on milestones for the protocol
### Description
Today we have an ad hoc way of working when it comes to milestone handling in the Eiffel protocol repository, where the milestones for each upcoming edition is filled with issues intended for that edition or already merged on to master and thus will be part of the edition.

### Motivation
It's better to have processes described in text than having them ad hoc :)

### Exemplification
See https://github.com/eiffel-community/eiffel/pull/277#issuecomment-931022248

### Benefits
Less risk of missing creating milestones and updating them

### Possible Drawbacks
None

### PR DESCRIPTION
**Pull request by magnusbaeck**
### Applicable Issues
Fixes #279 

### Description of the Change
Adds documentation on the use of milestones and an edition release checklist.

### Alternate Designs
None.

### Benefits
Less mental burden when releasing an edition and decreased risk of forgetting anythign.

### Possible Drawbacks
None.

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I    have the right to submit it under the open source license    indicated in the file; or

(b) The contribution is based upon previous work that, to the best    of my knowledge, is covered under an appropriate open source    license and I have the right under that license to submit that    work with modifications, whether created in whole or in part    by me, under the same open source license (unless I am    permitted to submit under a different license), as indicated    in the file; or

(c) The contribution was provided directly to me by some other    person who certified (a), (b) or (c) and I have not modified    it.

(d) I understand and agree that this project and the contribution    are public and that a record of the contribution (including all    personal information I submit with it, including my sign-off) is    maintained indefinitely and may be redistributed consistent with    this project or the open source license(s) involved.

Signed-off-by: Magnus Bäck \<magnus.back@axis.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -86,3 +86,5 @@ Visit [Eiffel Community](https://eiffel-community.github.io) to get started and
    1. [Custom Data](./customization/custom-data.md)
 1. Extensions
    1. [Eiffel Operations Extension](https://github.com/Ericsson/eiffel-operations-extension)
+1. Maintenance
```
Comment (magnusbaeck) [inline README.md:89]: Not sure Maintenance is the best choice of word but that's what I could come up with at this point.

File: eiffel-syntax-and-usage/event-schemas.md [hunk 2]
```diff
@@ -35,7 +35,7 @@ Both schemas and documentation files are generated from _schema definition files
 | `_links.<link type>.targets.types`     | A string array of event names that the link type may point to. Must be non-empty if `any_type` is false, and must be empty if `any_type` is true. |
 | `_history`      | An array of objects describing the event type's version history, up to and including the current version. The items should be sorted in reverse order. |
 | `_history.version`        | The event version described in this item. |
-| `_history.introduced_in`  | A description of the edition in which this item's event version was introduced. |
+| `_history.introduced_in`  | The first edition in which this version was the latest for this event type. If the version hasn't been released in a protocol edition or if there are newer versions in the first edition where it was introduced, use "No edition set".|
```
Comment (magnusbaeck) [inline eiffel-syntax-and-usage/event-schemas.md:38]: This text is completely crazy but it's unfortunately an accurate description of how this field is used. I'm going to file an issue that revamps how we use this and/or the `introduced_in` name.

File: README.md (hunk 1 above)
Comment (e-backmark-ericsson) [reply README.md:89]: I'm not sure it's much better, but maybe more to the point anyway.
```suggestion
1. Releases and Milestones
```

**Review by e-backmark-ericsson — CHANGES_REQUESTED** [review]
Provided a different suggestion on one of the headings.
Also waiting for the "other findings" to be documented as issues before closing #279

File: eiffel-syntax-and-usage/event-schemas.md (hunk 2 above)
Comment (magnusbaeck) [reply eiffel-syntax-and-usage/event-schemas.md:38]: Issue filed: #326

File: README.md (hunk 1 above)
Comment (magnusbaeck) [reply README.md]: Yeah, that's probably better. Done.

File: releases-and-milestones/release-process.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+## Making a release
+
+When all issues in the scope of the edition have been closed and it's time to make the release, follow these steps:
+
+1.  Verify that all issues in the [milestone](https://github.com/eiffel-community/eiffel/milestones?state=open) are closed.
+1.  Create a pull request with the following changes (see [PR 277](https://github.com/eiffel-community/eiffel/pull/277) for reference):
+    1.  Change all "No edition set" entries in the release tables of the latest YAML file for each event type to a link to the not yet existing edition tag.
+    1.  Claim the edition in [versioning.md](../eiffel-syntax-and-usage/versioning.md), including a short summary of the changes in the edition.
+1.  When the pull request has been merged, create and push an "edition-\<name>" annotated tag (use `git tag -a`). The tag comment could include a short version of the included changes to the protocol. Any new major versions of event types should be called out.
+1.  Create a GitHub release based on the newly pushed tag. The tag comment can probably be reused as the release description.
```
Comment (m-linner-ericsson) [inline releases-and-milestones/release-process.md:37]: Are there any settings when creating the release e.g. https://github.com/eiffel-community/eiffel-remrem-publish/releases has different layout of the release. Any tick boxes you need to mark?

Comment (e-backmark-ericsson) [reply releases-and-milestones/release-process.md:37]: They are very similar to me: https://github.com/eiffel-community/eiffel/releases vs https://github.com/eiffel-community/eiffel-remrem-publish/releases. What are the differences you refer to, @m-linner-ericsson ?

**Comment by e-backmark-ericsson** [comment]
I'm ok with your latest changes @magnusbaeck .
Now I just want to see that the "other findings" are documented as issues before approving it (since it automatically closes #279)

**Review by e-backmark-ericsson — APPROVED** [review]
Sorry, I missed your references in the linked issue. Approved.

File: releases-and-milestones/release-process.md (hunk 3 above)
Comment (m-linner-ericsson) [reply releases-and-milestones/release-process.md:37]: image

The red circled parts are missing in 2.0.26

Comment (e-backmark-ericsson) [reply releases-and-milestones/release-process.md:37]: It seems that it is possible to generate a nice looking release note once a tag is selected, and that release note will include the Contributors and maybe more interesting stuff.
I propose that we leave the current instruction as-is, and then once we have tested this feature along with the upcoming release we could possibly update the instruction to recommend/require that the release notes are generated using that button.

Comment (magnusbaeck) [reply releases-and-milestones/release-process.md:37]: Interesting, I hadn't seen that feature. Let's try it out for Arica. There are ways of [configuring what PRs to include](https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes#configuring-automatically-generated-release-notes) and that's something we could use to only include PRs affecting the protocol (if we start labeling issues and PRs with e.g. a "protocol" label, but I think we should do that anyway).

**Review by t-persson — APPROVED** [review]

**Pull request merged**